cmplibrary/src/test/java/com/sourcepoint/cmplibrary/SourcePointClientBuilderTest.java

```diff
@@ -0,0 +1,72 @@
+package com.sourcepoint.cmplibrary;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+public class SourcePointClientBuilderTest {
+    SourcePointClientBuilder sourcePointClientBuilder;
+
+    @Before
+    public void setUp() throws ConsentLibException {
+        sourcePointClientBuilder = new SourcePointClientBuilder(808, "siteName", true);//mock(SourcePointClientBuilder.class);
+    }
+
+
+    @Test
+    public void setMmsDomain() throws Exception{
+        sourcePointClientBuilder.setMmsDomain("mmsDomain");
+        Field member = null;
+
+        member = SourcePointClientBuilder.class.getDeclaredField("mmsDomain");
+
+        member.setAccessible(true);
+        SourcePointClientBuilder sourcePointClientBuilder1 = sourcePointClientBuilder;
+
+        assertEquals("mmsDomain", member.get(sourcePointClientBuilder1).toString());
```

Comment:
The code to get set a value, turn its member field public and asserting its value is repeated 3 times. Maybe we could refactor it to avoid duplication. If we use my suggestion above, we could write a method like:

```java private Field getPublicField(String fieldName, Object subject) {
  Field field = subject.class.getDeclaredField(fieldName);
  member.setAccessible(true);
  return field;
```

Then, in our tests:
```java
    @Test
    public void setCmpDomain() throws Exception{
        SourcePointClientBuilder clientBuilder = defaultClientBuilder()
        clientBuilder.setMmsDomain("mmsDomain");

        assertEquals("mmsDomain", getPublicField("mmsDomain", clientBuilder).toString());
    }
```

What you think?

Reply:
will work on this.

Reply:
implemented as suggested